runtime/compilers/rillv1/parse_metrics_view.go

```diff
@@ -171,6 +165,74 @@ func (p *Parser) parseMetricsView(ctx context.Context, node *Node) error {
 		return fmt.Errorf("must define at least one measure")
 	}
 
+	if tmp.Policy != nil {
+		templateData := TemplateData{User: map[string]interface{}{
+			"name":   "dummy",
+			"email":  "mock@example.org",
+			"domain": "example.org",
+			"groups": []interface{}{"all"},
+			"admin":  false,
+		}}
+
+		if tmp.Policy.HasAccess != "" {
+			hasAccess, err := ResolveTemplate(tmp.Policy.HasAccess, templateData)
+			if err != nil {
+				return fmt.Errorf(`invalid 'policy': 'has_access' templating is not valid: %w`, err)
+			}
+			_, err = EvaluateBoolExpression(hasAccess)
+			if err != nil {
+				return fmt.Errorf(`invalid 'policy': 'has_access' expression not valuating to a boolean: %w`, err)
+			}
+		}
+
+		if tmp.Policy.Filter != "" {
+			_, err := ResolveTemplate(tmp.Policy.Filter, templateData)
+			if err != nil {
+				return fmt.Errorf(`invalid 'policy': 'filter' templating is not valid: %w`, err)
+			}
+		}
+
+		if len(tmp.Policy.Include) > 0 && len(tmp.Policy.Exclude) > 0 {
+			return errors.New("invalid 'policy': only one of 'include' and 'exclude' can be specified")
+		}
+		if tmp.Policy.Include != nil {
+			for _, include := range tmp.Policy.Include {
+				if include.Name == "" || include.Condition == "" {
+					return fmt.Errorf("invalid 'policy': 'include' fields must have a valid 'name' and 'if' condition")
+				}
+				if !names[include.Name] {
+					return fmt.Errorf("invalid 'policy': 'include' property %q does not exists in dimensions list", include.Name)
```

Comment:
1. `names` covers both dimension and measure names – so I guess the error should be "does not match a dimension or measure name"
2. Since the validation now checks both dimension and measure names, we should update the validation in the API handlers to check measure names as well (and also update the `GetCatalogEntry` call to filter out inaccessible measures)